From 48078b9f342c6b834c07b6c553ff3919ccde8021 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Wed, 4 Dec 2013 17:56:32 +0100
Subject: [LIBREPORT PATCH] make_description: add an option for URLs from
 reported_to

The option causes that the description text will contain all URLs found
in reported_to file.

For reported_to line:
Bugzilla: URL=http://bugzilla.redhat.com/1000000

the following line will be included in the description:
Bugzilla: http://bugzilla.redhat.com/1000000

Related to rhbz#1036585

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/include/internal_libreport.h |  2 ++
 src/lib/make_descr.c             | 29 +++++++++++++++++++++++++++++
 2 files changed, 31 insertions(+)

diff --git a/src/include/internal_libreport.h b/src/include/internal_libreport.h
index 6cfbf26..59076a1 100644
--- a/src/include/internal_libreport.h
+++ b/src/include/internal_libreport.h
@@ -581,6 +581,8 @@ enum {
     MAKEDESC_SHOW_MULTILINE = (1 << 1),
     MAKEDESC_SHOW_ONLY_LIST = (1 << 2),
     MAKEDESC_WHITELIST      = (1 << 3),
+    /* Include all URLs from FILENAME_REPORTED_TO element in the description text */
+    MAKEDESC_SHOW_URLS      = (1 << 4),
 };
 #define make_description libreport_make_description
 char *make_description(problem_data_t *problem_data, char **names_to_skip, unsigned max_text_size, unsigned desc_flags);
diff --git a/src/lib/make_descr.c b/src/lib/make_descr.c
index a4e7938..2316681 100644
--- a/src/lib/make_descr.c
+++ b/src/lib/make_descr.c
@@ -101,6 +101,35 @@ char *make_description(problem_data_t *problem_data, char **names_to_skip,
     }
 
     bool append_empty_line = !empty;
+    if (desc_flags & MAKEDESC_SHOW_URLS)
+    {
+        const char *reported_to = problem_data_get_content_or_NULL(problem_data, FILENAME_REPORTED_TO);
+        if (reported_to != NULL)
+        {
+            GList *reports = read_entire_reported_to_data(reported_to);
+
+            for (GList *iter = reports; iter != NULL; iter = g_list_next(iter))
+            {
+                const report_result_t *const report = (report_result_t *)iter->data;
+
+                if (report->url == NULL)
+                    continue;
+
+                if (append_empty_line)
+                    strbuf_append_char(buf_dsc, '\n');
+                append_empty_line = false;
+                empty = false;
+
+                int pad = 16 - (strlen(report->label) + 2);
+                if (pad < 0) pad = 0;
+                strbuf_append_strf(buf_dsc, "%s: %*s%s\n", report->label, pad, "", report->url);
+            }
+
+            g_list_free_full(reports, (GDestroyNotify)free_report_result);
+        }
+    }
+
+    append_empty_line = !empty;
     if (desc_flags & MAKEDESC_SHOW_FILES)
     {
         /* Print file info. Format:
-- 
1.8.3.1

